File: hackerrank/Algorithms/search/pairs.py
"""
  @file pairs.py
  @author Rahul Awasthi (awasthir@pm.me)
  
  @brief 
  try out the problem here:
    https://www.hackerrank.com/challenges/pairs/
    
"""

def pairs(k, arr):
    # Write your code here
    arr = set(arr)
    return sum(1 for i in arr if i+k in arr)

File: hackerrank/Algorithms/search/pairs.cpp
/**
 * @file pairs.cpp
 * @author Rahul Awasthi (awasthir@pm.me)
 * 
 * @brief 
 * problem link:
 * https://www.hackerrank.com/contests/h-cs3c-feb16/challenges/pairs
 * 
 */

int pairs(int k, vector<int> arr) {
    set<int> s(arr.begin(), arr.end());
    int count = 0;
    for (auto i : s) {
        if (s.find(i+k) != s.end()) count++;
        if (s.find(i-k) != s.end()) count++;
    }
    return count/2;
}